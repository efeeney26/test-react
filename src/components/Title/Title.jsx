import React from 'react'
import PropTypes from 'prop-types'

const Title = ({ title }) => (
  title
    ? <h1>
      {title}
    </h1>
    : <p>Default title</p>
)

Title.propTypes = {
  title: PropTypes.string
}

Title.defaultProps = {
  title: ''
}

export default Title
