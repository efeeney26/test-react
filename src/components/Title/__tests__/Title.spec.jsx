import React from 'react'
import { shallow } from 'enzyme'

import Title from '../Title'

describe('Title', () => {
  const component = shallow(<Title />)

  it('should render with default title', () => {
    expect(component).toMatchSnapshot()
  })
})

describe('Title', () => {
  const props = {
    title: 'hello'
  }

  const component = shallow(<Title {...props}/>)

  it('should render with title', () => {
    expect(component).toMatchSnapshot()
  })
})
