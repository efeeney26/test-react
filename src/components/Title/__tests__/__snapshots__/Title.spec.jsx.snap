// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render with default title 1`] = `
<p>
  Default title
</p>
`;

exports[`Title should render with title 1`] = `
<h1>
  hello
</h1>
`;
